src/main/java/art/heredium/domain/coupon/repository/CouponUsageRepository.java

```diff
@@ -1,9 +1,25 @@
 package art.heredium.domain.coupon.repository;
 
+import java.util.List;
+
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
+import art.heredium.domain.coupon.entity.Coupon;
 import art.heredium.domain.coupon.entity.CouponUsage;
 
 @Repository
-public interface CouponUsageRepository extends JpaRepository<CouponUsage, Long> {}
+public interface CouponUsageRepository extends JpaRepository<CouponUsage, Long> {
+
+  @Query("SELECT DISTINCT cu.coupon FROM CouponUsage cu WHERE cu.account.id = :accountId")
+  List<Coupon> findDistinctCouponsByAccountId(@Param("accountId") Long accountId);
+
+  @Query(
+      "SELECT cu FROM CouponUsage cu WHERE cu.account.id = :accountId AND cu.coupon.id = :couponId AND cu.isUsed = :isUsed")
+  List<CouponUsage> findByAccountIdAndCouponIdAndIsUsed(
+      @Param("accountId") Long accountId,
+      @Param("couponId") Long couponId,
+      @Param("isUsed") boolean isUsed);
+}
```

Comment:
```suggestion
  List<CouponUsage> findByAccountIdAndCouponIdAndIsUsedTrue(
      @Param("accountId") Long accountId,
      @Param("couponId") Long couponId);
}

 List<CouponUsage> findByAccountIdAndCouponIdAndIsUsedFalse(
      @Param("accountId") Long accountId,
      @Param("couponId") Long couponId);
}
```
Thử 2 cái này xem. Nếu được thì đi theo hướng này, khỏi phải viết @Query, và cũng ko cần truyền thêm param isUsed

Reply:
I updated in https://github.com/Incuvers-IT/heredium-BE/pull/9/commits/d285cb6158ae285e4a06c5e9bfff3eec0f9a967a